aws-cpp-sdk-core/source/utils/threading/Executor.cpp

```diff
@@ -107,3 +107,58 @@ bool PooledThreadExecutor::HasTasks()
     std::lock_guard<std::mutex> locker(m_queueLock);
     return m_tasks.size() > 0;
 }
+
+BlockingExecutor::BlockingExecutor(std::shared_ptr<Executor> executor, size_t poolSize) :
+m_executor(executor), m_poolSize(poolSize), m_numTasksRunning(0)
+{
+}
+
+BlockingExecutor::~BlockingExecutor()
+{
+    while (m_numTasksRunning != 0)

```

Comment:
I typically prefer to have a public method on things like this to allow the user to call shutdown independently of the destruction of the object. The destructor should certainly call that shutdown function, but it gives the user more control over where they take the hit.